src/main/java/io/jenkins/plugins/artifact_manager_jclouds/JCloudsArtifactManager.java

```diff
@@ -136,17 +136,21 @@ public boolean delete() throws IOException, InterruptedException {
      * Delete all blobs starting with prefix
      */
     public static boolean delete(BlobStoreProvider provider, BlobStore blobStore, String prefix) throws IOException, InterruptedException {
-        Iterator<StorageMetadata> it = new JCloudsVirtualFile.PageSetIterable(blobStore, provider.getContainer(), ListContainerOptions.Builder.prefix(prefix).recursive());
-        boolean found = false;
-        while (it.hasNext()) {
-            StorageMetadata sm = it.next();
-            String path = sm.getName();
```

Comment:
To see the change more clearly: [diff](https://github.com/jenkinsci/artifact-manager-s3-plugin/pull/32/files?w=1)